Source/SDK/ClientBuilder.cs

```diff
@@ -178,7 +192,7 @@ public Client Build()
                 Claims.Empty,
                 CultureInfo.InvariantCulture);
 
-            var eventTypes = _eventTypesBuilder.Build();
+            var eventTypes = _eventTypesBuilder.Build(new EventTypes(_loggerFactory.CreateLogger<EventTypes>()));
```

Comment:
It feels a bit strange that it takes in an `EventTypes` and returns an `EventTypes`, it makes it a bit unclear if it modifies the thing you pass in, or if it creates a brand new one. I suggest we rename the method to something like `AddAssociationsInto`, and not don't return anything from it.

Reply:
This should be resolved by latest commits